src/endpoint.ts

```diff
@@ -1,26 +1,37 @@
-import { ArchethicRPCClient } from "./api/wallet_rpc.js";
+import { ConnectionState } from "./api/types.js";
+import { AWCWebBrowserExtension } from "./api/wallet_rpc.browserextension.js";
+import { ArchethicWalletClient } from "./api/wallet_rpc.js";
+import { AWCWebsocketStreamChannel } from "./api/wallet_rpc.websocket.js";
+
+export abstract class Endpoint {
+  abstract get isRpcAvailable(): boolean;
+  abstract get origin(): string;
+  abstract get nodeEndpoint(): URL | null;
 
-export class Endpoint {
   /**
-   * @param {String} endpoint
-   * @return {DirectEndpoint | WalletRPCEndpoint}
+   * @param {String | undefined} endpoint
+   * @return {Endpoint}
    */
-  static build(endpoint: string): DirectEndpoint | WalletRPCEndpoint {
-    const url: URL = new URL(endpoint);
-
-    if (url.protocol === "ws:") {
-      return new WalletRPCEndpoint(endpoint);
+  static build(endpoint: string | undefined): Endpoint {
+    if (endpoint === undefined) {
+      console.log('Using AWC client');
+      return new AWCEndpoint(
+        AWCWebBrowserExtension.awc ??
+        new ArchethicWalletClient(new AWCWebsocketStreamChannel(`ws://localhost:12345`))
+      )
     }
 
+    const url: URL = new URL(endpoint);
     if (url.protocol === "http:" || url.protocol === "https:") {
+      console.log('Using direct endpoint');
       return new DirectEndpoint(endpoint);
     }
 
     throw new Error(`Invalid endpoint protocol: ${url.protocol}`);
   }
 }
 
-export class DirectEndpoint {
+export class DirectEndpoint implements Endpoint {
```

Comment:
Technically, `Endpoint` is an `abstract class` (and not an `interface`) because it implements a `static` method (`build`).

That being said, `Endpoint` cumulates two responsabilities : it is both a `factory`, and an `interface`. 
Maybe we should split `Endpoint` in two (`Endpoint` and `EndpointFactory`).

Reply:
@bchamagne done